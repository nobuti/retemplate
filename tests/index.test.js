import React from 'react'
import renderer from 'react-test-renderer'

import template from '../src'

describe('template', () => {
  it('should return an array of children', () => {
    expect(Array.isArray(template('wadus'))).toBe(true)
  })

  it('should work without data', () => {
    const component = renderer.create(<div>{template('hello')}</div>)

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should replace string with data', () => {
    const component = renderer.create(<div>{template('hello {world}', {
      world: 'girl'
    })}</div>)

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should replace string with data function result', () => {
    const component = renderer.create(<div>{template('good {time}', {
      time: () => 'morning'
    })}</div>)

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should replace string with data function result', () => {
    const component = renderer.create(<div>{template('hello {world}', {
      world: () => <span>world</span>
    })}</div>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
