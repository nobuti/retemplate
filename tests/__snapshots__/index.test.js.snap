// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template should replace string with data 1`] = `
<div>
  hello 
  girl
</div>
`;

exports[`template should replace string with data function result 1`] = `
<div>
  good 
  morning
</div>
`;

exports[`template should replace string with data function result 2`] = `
<div>
  hello 
  <span>
    world
  </span>
</div>
`;

exports[`template should work without data 1`] = `
<div>
  hello
</div>
`;
